lib/stones-spec/precondition.rb

```diff
@@ -2,9 +2,9 @@ module StonesSpec
   class Precondition
     attr_reader :initial_board_gbb
 
-    def initialize(initial_board_gbb, arguments)
-      @initial_board_gbb = initial_board_gbb
-      @arguments = arguments
+    def initialize(example)
+      @initial_board_gbb = example.initial_board
```

Comment:
It is not a great idea to pass the example to the precondition, since it ties creating a `Precondition` to having an `Example` like object. 

As a rule of thumb, just use constructors to inyect dependencies. Use a class method like `from_example` instead